File: myactivities/src/main/java/pl/mareklangiewicz/myactivities/MyTestActivity.java
package pl.mareklangiewicz.myactivities;

import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.annotation.Nullable;
import android.view.MenuItem;
import android.view.View;
import android.view.animation.LinearInterpolator;

import pl.mareklangiewicz.mydrawables.MyLivingDrawable;
import pl.mareklangiewicz.mydrawables.MyMagicLinesDrawable;
import pl.mareklangiewicz.myviews.IMyUINavigation;

import static android.animation.ObjectAnimator.ofInt;
import static android.animation.ObjectAnimator.ofPropertyValuesHolder;
import static android.animation.PropertyValuesHolder.ofFloat;
import static android.view.View.ALPHA;
import static android.view.View.TRANSLATION_Y;

/**
 * Test activity presenting most of the MyBlocks functionality
 */

public final class MyTestActivity extends pl.mareklangiewicz.myactivities.MyActivity {

    private final MyLivingDrawable mMyMagicLinesDrawable = new MyMagicLinesDrawable();
    private @Nullable ObjectAnimator mHomePageAnimator;
    private @Nullable ObjectAnimator mMagicLinesAnimator;


    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        log.i("Hello world!");
        log.d("some boring debug message...");
        log.w("Warning!... just kidding...");

        //noinspection ConstantConditions
        getGnav().inflateMenu(R.menu.ma_my_test_global);
        getGnav().inflateHeader(R.layout.ma_my_test_global_header);

        //noinspection ConstantConditions
        View underline = getGnav().getHeader().findViewById(R.id.ma_mta_gh_v_underline);
        mMyMagicLinesDrawable.setColor(0x30ffffff);
        mMyMagicLinesDrawable.setStrokeWidth(dp2px(4));
        underline.setBackground(mMyMagicLinesDrawable);

        View homepage = getGnav().getHeader().findViewById(R.id.ma_mta_gh_tv_home_page);

        PropertyValuesHolder pvha = ofFloat(ALPHA, 0f, 0f, 1f);
        PropertyValuesHolder pvhy = ofFloat(TRANSLATION_Y, -50f, -50f, 0f);

        mHomePageAnimator = ofPropertyValuesHolder(homepage, pvha, pvhy);
        mHomePageAnimator.setInterpolator(new LinearInterpolator());

        mMagicLinesAnimator = ofInt(mMyMagicLinesDrawable, "level", 0, 10000);
        mMagicLinesAnimator.setDuration(1000).setInterpolator(new LinearInterpolator());

        if(savedInstanceState == null)
            getGnav().setCheckedItem(R.id.section_my_pie_tests, true);
    }

    @Override public void onDrawerSlide(View drawerView, float slideOffset) {
        super.onDrawerSlide(drawerView, slideOffset);
        if(drawerView != mGlobalNavigationView)
            return;
        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
            if(mHomePageAnimator != null)
                mHomePageAnimator.setCurrentFraction(slideOffset);
        }
    }


    @Override public void onDrawerOpened(View drawerView) {
        super.onDrawerOpened(drawerView);
        if(drawerView != mGlobalNavigationView)
            return;
        if(mMagicLinesAnimator != null)
            if(!mMagicLinesAnimator.isStarted())
                mMagicLinesAnimator.start();
    }

    @Override public void onDrawerClosed(View drawerView) {
        super.onDrawerClosed(drawerView);
        if(drawerView != mGlobalNavigationView)
            return;
        if(mMagicLinesAnimator != null)
            mMagicLinesAnimator.cancel();
        mMyMagicLinesDrawable.setLevel(0);
    }

    @Override public boolean onItemSelected(IMyUINavigation nav, MenuItem item) {

        boolean done = super.onItemSelected(nav, item);

        if(done)
            return true;

        @IdRes int id = item.getItemId();

        if(id == R.id.action_whats_up) {
            log.i("[SNACK][SHORT]What's up mate?");
            return true;
        }
        else if(id == R.id.action_settings) {
            log.w("[SNACK]TODO: some settings (or not)..");
            return true;
        }
        else if(id == R.id.action_destroy_something) {
            log.a("[SNACK]BUM!"); /* throw new InternalError("BUM!"); */
            return true;
        }

        return false;
    }

}

File: myactivities/src/main/java/pl/mareklangiewicz/myactivities/MyTestActivity.kt
package pl.mareklangiewicz.myactivities

import android.animation.ObjectAnimator
import android.animation.ObjectAnimator.ofInt
import android.animation.ObjectAnimator.ofPropertyValuesHolder
import android.animation.PropertyValuesHolder
import android.annotation.TargetApi
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.animation.LinearInterpolator
import kotlinx.android.synthetic.main.ma_my_test_global_header.view.*
import pl.mareklangiewicz.mydrawables.MyMagicLinesDrawable
import pl.mareklangiewicz.myutils.*

/**
 * Test activity presenting most of My..... classes functionality
 */
open class MyTestActivity : MyActivity() {

    private val mldrawable = MyMagicLinesDrawable().apply {
        color = 0x30ffffff
        strokeWidth = 8f
    }

    private val mlanimator = ofInt(mldrawable, "level", 0, 10000).apply {
        duration = 1000
        interpolator = LinearInterpolator()
    }

    private lateinit var hpanimator: ObjectAnimator

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        log.i("Hello world!")
        log.d("some boring debug message...")
        log.w("Warning!... just kidding...")

        gnav!!.menuId = R.menu.ma_my_test_global
        gnav!!.headerId = R.layout.ma_my_test_global_header

        gnav!!.headerObj!!.ma_mta_gh_v_underline.background = mldrawable

        hpanimator = ofPropertyValuesHolder(
                gnav!!.headerObj!!.ma_mta_gh_tv_home_page,
                PropertyValuesHolder.ofFloat(View.ALPHA, 0f, 0f, 1f),
                PropertyValuesHolder.ofFloat(View.TRANSLATION_Y, -50f, -50f, 0f)
        ).apply { interpolator = LinearInterpolator() }

        gnav!!.items { // we ignore returned subscription - navigation will live as long as activity
            when(it) {
                R.id.action_whats_up -> log.i("[SNACK][SHORT]What's up mate?")
                R.id.action_settings -> log.w("[SNACK]TODO: some settings (or not)..")
                R.id.action_destroy_something -> log.a("[SNACK]BUM!")
            }
        }

        if (savedInstanceState == null)
            gnav!!.setCheckedItem(R.id.section_my_pie_tests, true)
    }

    @TargetApi(22)
    override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
        super.onDrawerSlide(drawerView, slideOffset)
        if (drawerView === gnav && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1)
            hpanimator.setCurrentFraction(slideOffset)
    }


    override fun onDrawerOpened(drawerView: View) {
        super.onDrawerOpened(drawerView)
        if (drawerView === gnav) mlanimator?.run { if (!isStarted) start() }
    }

    override fun onDrawerClosed(drawerView: View) {
        super.onDrawerClosed(drawerView)
        if (drawerView === gnav) {
            mlanimator?.cancel()
            mldrawable.level = 0
        }
    }

}
